### PR TITLE
Issue certs with unique ids

### LINKED ISSUE #268: Issue certs with unique id numbers
These should be within the 100000-Inf range and preferrably persisted in the database. 

` cert_number = str(random.randint(100000, Inf))`

DoD: Whenever a new digital certificate is issued, the individual will be assigned an immutable and unique certificate number.

### PR DESCRIPTION
**Pull request by jonandernovella**
- Used a SQL sequence for auto generation of unique ids within a series.
- Official certificate numbers become the digital id when issuing a digital certificate.

Closes #268

### CODE REVIEW

File: app/utils/database.py [hunk 1]
```diff
@@ -398,6 +398,8 @@ class Meta:  # pylint: disable=too-few-public-methods
 
         indexes = ((("mother", "father", "birth_date"), True),)
 
+## Create sequence to allow unique ids for digital certificates
+DATABASE.execute_sql('''CREATE SEQUENCE IF NOT EXISTS certificates_seq START WITH 100000 INCREMENT BY 1 MAXVALUE 199999 NO CYCLE''')
```
Comment (pontus) [inline app/utils/database.py]: Move this to `init`?

I think sqlite doesn't support `CREATE SEQUENCE`, so only run if using postgres. 

(How does this match with tests?)

File: app/utils/database.py [hunk 2]
```diff
@@ -410,6 +412,7 @@ class Individual(BaseModel):
     origin_herd = ForeignKeyField(Herd)
     name = CharField(50, null=True)
     certificate = CharField(20, null=True)
+    digital_certificate = IntegerField(sequence="certificates_seq", unique=True)
```
Comment (pontus) [inline app/utils/database.py]: Allow `null` (e.g. with `null=True`)?

**Comment by pontus** [comment]
Migration?

**Comment by pontus** [comment]
Maybe I'm not following along as I need to - does "Official certificate numbers become the digital id when issuing a digital certificate." mean that we copy the value into `certificate`?

If not - can we not separate individuals without certificate from those who have one from the database only?

File: app/utils/database.py (hunk 2 above)
Comment (jonandernovella) [reply app/utils/database.py]: done

**Comment by jonandernovella** [comment]
> Maybe I'm not following along as I need to - does "Official certificate numbers become the digital id when issuing a digital certificate." mean that we copy the value into `certificate`?
> 
> If not - can we not separate individuals without certificate from those who have one from the database only?

Yes, we do copy the value into certificate.

File: app/utils/database.py (hunk 1 above)
Comment (jonandernovella) [reply app/utils/database.py]: It should be now in init(). I am not sure what to do about sqlite.

**Comment by pontus** [comment]
> > Maybe I'm not following along as I need to - does "Official certificate numbers become the digital id when issuing a digital certificate." mean that we copy the value into `certificate`?
> > If not - can we not separate individuals without certificate from those who have one from the database only?
> 
> Yes, we do copy the value into certificate.

Ok, good, do you recall where this happens (I don't see it quickly)?

File: app/herdbook.py [hunk 3]
```diff
@@ -531,9 +532,10 @@ def issue_certificate(i_number):
...
         return jsonify({"response": "Certificate already exists"}), 400
 
-    ##cert_number = str(random.randint(50001, 99999))
-    cert_number = "99999"
-    ind.update(**request.json, certificate=cert_number)
+    form = request.json
+    cert_number = ind["digital_certificate"]
+
+    ind.update(**form, certificate=cert_number)
     da.update_individual(ind, session.get("user_id", None))
```
Comment (pontus) [inline app/herdbook.py]: Same as https://github.com/NBISweden/theherdbook/pull/251/files#r635138222

File: app/utils/database.py [hunk 4]
```diff
@@ -410,6 +409,7 @@ class Individual(BaseModel):
     origin_herd = ForeignKeyField(Herd)
     name = CharField(50, null=True)
     certificate = CharField(20, null=True)
+    digital_certificate = IntegerField(sequence="certificates_seq", unique=True, null=True)
```
Comment (pontus) [inline app/utils/database.py:412]: I think this needs to be fixed to work with test as well, either by moving test to postgres or supporting sqlite, but I don't know exactly how yet.

I don't see that it needs to delay this specific PR, though (but should be fixed before merge to develop).

File: app/utils/database.py [hunk 5]
```diff
@@ -1091,6 +1091,9 @@ def init():
     with DATABASE.atomic():
         sh_bootstrap.save()
 
+    ## Create sequence to allow unique ids for digital certificates
+    DATABASE.execute_sql('''CREATE SEQUENCE IF NOT EXISTS certificates_seq START WITH 100000 INCREMENT BY 1 MAXVALUE 199999 NO CYCLE''')
```
Comment (pontus) [inline app/utils/database.py]:
```suggestion
    if type(DATABASE) == PostgresqlDatabase:
        DATABASE.execute_sql('''CREATE SEQUENCE IF NOT EXISTS certificates_seq START WITH 100000 INCREMENT BY 1 MAXVALUE 199999 NO CYCLE''')
```

**Review by pontus — APPROVED** [review]

**Pull request merged**